lib/apiservers/engine/backends/image.go

```diff
@@ -147,8 +155,22 @@ func (i *Image) PullImage(ctx context.Context, ref reference.Named, metaHeaders
 		return err
 	}
 
-	client := PortLayerClient()
-	ImageCache().Update(client)
+	imageJSON, err := ioutil.ReadFile(imageFile)
```

Comment:
@jzt i think in the multiwriter case, (please correct me if I'm wrong), you'd have to deal with alignment of the stream yourself.  It's not impossible, but it's not trivial either.